operator/config/manager/manager.yaml

```diff
@@ -47,3 +47,5 @@ spec:
       terminationGracePeriodSeconds: 10
       securityContext:
         runAsNonRoot: true
+        seccompProfile:
+           type: RuntimeDefault
```

Comment:
This change should be addressed on the overlays for `community-openshift` and `openshift` to skip pulling this change into the `community` bundle. Latter is dedicated to vanillia k8s clusters and we cannot assume that they support this profile.

Reply:
Shouldn't then we also be moving `runAsNonRoot` into those bundles as well, @periklis ? Setting this value assumes the operator running with the Kubernetes's `restricted` pod security standard. Setting the `seccompProfile` is mandatory for that standard.

https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted

Reply:
I don't recall why we kept them though? Can you research a bit if we missed something back then?

Reply:
I think this may have been overlooked with the [original implementation](https://github.com/grafana/loki/pull/6514). 

We smoke tested to ensure it would work in vanilla K8s, but I don't think we considered if needed to be `restricted` compliant there. OpenShift 4.11+ requires it, but don't see any reason why we need to enforce it on the vanilla K8s ones.